src/lib.rs

```diff
@@ -22,104 +63,192 @@ impl fmt::Display for DeviceType {
-    pub maximum_temperature_in_kelvin: u16,
+/// A device-relatred error.
+#[derive(Debug, Clone, PartialEq)]
+pub enum DeviceError {
+    /// Tried to use a device that is not supported.
+    Unsupported,
 }
 
-const VENDOR_ID: u16 = 0x046d;
-const PRODUCT_IDS: [u16; 4] = [0xc900, 0xc901, 0xb901, 0xc903];
-const USAGE_PAGE: u16 = 0xff43;
+/// A device that can be used.
+#[derive(Debug)]
+pub struct Device<'a> {
+    device_info: &'a DeviceInfo,
+    device_type: DeviceType,
+}
 
-fn get_device_type(product_id: u16) -> DeviceType {
-    match product_id {
-        0xc900 => DeviceType::LitraGlow,
-        0xc901 => DeviceType::LitraBeam,
-        0xb901 => DeviceType::LitraBeam,
-        0xc903 => DeviceType::LitraBeamLX,
-        _ => panic!("Unknown product ID"),
+impl<'a> TryFrom<&'a DeviceInfo> for Device<'a> {
+    type Error = DeviceError;
+
+    fn try_from(device_info: &'a DeviceInfo) -> Result<Self, DeviceError> {
+        if device_info.vendor_id() != VENDOR_ID || device_info.usage_page() != USAGE_PAGE {
+            return Err(DeviceError::Unsupported);
+        }
+        device_type_from_product_id(device_info.product_id())
+            .map(|device_type| Device {
+                device_info,
+                device_type,
+            })
+            .ok_or(DeviceError::Unsupported)
     }
 }
 
-fn get_minimum_brightness_in_lumen(device_type: &DeviceType) -> u16 {
-    match device_type {
-        DeviceType::LitraGlow => 20,
-        DeviceType::LitraBeam | DeviceType::LitraBeamLX => 30,
+impl Device<'_> {
+    /// The serial number of the device (if available).
+    #[must_use]
+    pub fn serial_number(&self) -> Option<&str> {
+        self.device_info.serial_number()
+    }
+
+    /// The model of the device.
+    #[must_use]
+    pub fn device_type(&self) -> DeviceType {
+        self.device_type
+    }
+
+    /// Opens the device and returns a [`DeviceHandle`] that can be used for getting and setting the
+    /// device status.
+    pub fn open(&self, api: &HidApi) -> HidResult<DeviceHandle> {
+        self.device_info
+            .open_device(api)
+            .map(|hid_device| DeviceHandle {
+                hid_device,
+                device_type: self.device_type,
+            })
     }
 }
 
-fn get_maximum_brightness_in_lumen(device_type: &DeviceType) -> u16 {
-    match device_type {
-        DeviceType::LitraGlow => 250,
-        DeviceType::LitraBeam | DeviceType::LitraBeamLX => 400,
+/// The handle of an opened device that can be used for getting and setting the device status.
+#[derive(Debug)]
+pub struct DeviceHandle {
+    hid_device: HidDevice,
+    device_type: DeviceType,
+}
+
+impl DeviceHandle {
+    /// The model of the device.
+    #[must_use]
+    pub fn device_type(&self) -> DeviceType {
+        self.device_type
+    }
+
+    /// Queries the current power status of the device. Returns `true` if the device is currently on.
+    pub fn is_enabled(&self) -> HidResult<bool> {
```

Comment:
I think I'd prefer to stick with the "on/off" paradigm rather than the "enabled" paradigm, because it matches more closely with the mental model of a light. What do you think?